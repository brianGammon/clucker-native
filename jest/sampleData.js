const sampleData1 = {
  userSettings: {
    key: 'user1',
    data: {
      currentFlockId: 'flock1',
    },
  },
  flocks: {
    data: {
      flock1: {
        name: 'Test Flock 1',
        ownedBy: 'user1',
      },
      flock2: {
        name: 'Test Flock 2',
        ownedBy: 'user2',
      },
    },
  },
  chickens: {
    data: {
      chicken1: {
        name: 'The First Chicken',
      },
      chicken2: {
        name: 'The Second Chicken',
      },
    },
  },
  eggs: {
    data: {
      egg1: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-01',
        modified: '2018-01-12T18:05:54.806Z',
        notes:
          'This was the record setting egg from May 2017, real egg was 66.1g on this day',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg2: {
        chickenId: 'chicken2',
        chickenName: 'Baby',
        damaged: false,
        date: '2017-09-24',
        modified: '2017-10-12T18:05:54.806Z',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 88.1,
      },
      egg3: {
        chickenId: 'chicken1',
        chickenName: 'Bossie',
        damaged: false,
        date: '2018-08-02',
        modified: '2018-08-02T18:05:54.806Z',
        notes: '',
        userId: 'MVQUEcoTaVUhUdo2ZS0zsIE6tn93',
        weight: 68.1,
      },
    },
  },
};

export default sampleData1;
