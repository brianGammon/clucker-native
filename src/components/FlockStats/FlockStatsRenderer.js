import React from 'react';
import { Text, ScrollView } from 'react-native';
import { type FlockStats as FlockStatsType } from '../../types';

type Props = {
  stats: FlockStatsType,
};

const FlockStatsRenderer = ({ stats }: Props) => {
  console.log('temp');
  return (
    <ScrollView>
      <Text>{JSON.stringify(stats, null, 2)}</Text>
    </ScrollView>
  );
};

export default FlockStatsRenderer;
