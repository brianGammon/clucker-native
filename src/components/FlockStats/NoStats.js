/* @flow */
import React from 'react';
import { Container, Text, Content } from 'native-base';
import HelpMessage from '../HelpMessage';
import Header from '../Header';

const NoStats = () => (
  <Container>
    <Header title="Flock Stats" eggButton />
    <Content padder>
      <HelpMessage>
        <Text>
          No eggs have been logged yet. Get started by tapping the &#34;+&#34;
          button above.
        </Text>
      </HelpMessage>
    </Content>
  </Container>
);

export default NoStats;
