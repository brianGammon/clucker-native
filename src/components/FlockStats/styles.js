import { StyleSheet } from 'react-native';
import variables from '../../styles/variables';

const label = {
  fontSize: 17,
  fontWeight: 'bold',
};

export default StyleSheet.create({
  eggStatsLabel: {
    ...label,
    alignSelf: 'center',
    marginBottom: 8,
  },
  eggStats: {
    flex: 1,
    flexDirection: 'row',
    justifyContent: 'space-between',
  },
  eggStatsCell: {
    alignItems: 'center',
  },
  eggStatsValue: {
    fontWeight: 'bold',
  },
  label: {
    ...label,
    color: variables.cluckerGrey,
  },
  subText: {
    fontSize: 14,
  },
  noEggsMessage: {
    flex: 1,
    alignItems: 'center',
  },
});
