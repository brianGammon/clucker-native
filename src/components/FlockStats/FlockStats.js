import React from 'react';
import { connect } from 'react-redux';
import FlockStatsRenderer from './FlockStatsRenderer';
import flockStatsSelector from '../../selectors/flockStatsSelector';
import currentFlockSelector from '../../selectors/currentFlockSelector';
import {
  type Navigation,
  type Flock,
  type FlockStats as FlockStatsType,
} from '../../types';

type Props = {
  navigation: Navigation,
  flock: Flock,
  stats: FlockStatsType,
};

class FlockStats extends React.Component<Props> {
  static navigationOptions = ({ navigation }) => ({
    title: navigation.getParam('title', 'Flock Stats'),
  });

  componentDidUpdate() {
    const { flock, navigation } = this.props;
    if (flock && flock.name) {
      if (navigation.getParam('title', 'default') !== flock.name) {
        navigation.setParams({ title: flock.name });
      }
    }
  }

  render() {
    const { stats } = this.props;
    return <FlockStatsRenderer stats={stats} />;
  }
}

const mapStateToProps = ({ userSettings, eggs, flocks }) => {
  let flock = {};
  if (userSettings.data.currentFlockId) {
    flock = currentFlockSelector(flocks.data, userSettings);
  }
  return {
    flock,
    stats: flockStatsSelector(eggs.data),
  };
};

export default connect(mapStateToProps)(FlockStats);
