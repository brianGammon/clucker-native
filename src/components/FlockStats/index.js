import FlockStats from './FlockStats';

export default FlockStats;
