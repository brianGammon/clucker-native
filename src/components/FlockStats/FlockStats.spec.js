import React from 'react';
import configureStore from 'redux-mock-store';
import sampleData from '../../../jest/test-data/full.2017-10-to-2018-05.json';
import FlockStats from './FlockStats';
import FlockStatsRenderer from './FlockStatsRenderer';

describe('FlockStats component:', () => {
  const middlewares = [];
  const mockStore = configureStore(middlewares);
  const store = mockStore(sampleData);

  test('FlockStats connected component renders', () => {
    const navigation = {};
    const wrapper = shallow(
      <FlockStats store={store} navigation={navigation} />,
    ).dive();
    expect(wrapper).toMatchSnapshot();
  });

  test('FlockStatsRenderer renders', () => {
    const mockStats = {
      total: 100,
      heaviest: null,
      averageWeight: 60,
      averageNumber: 2,
      mostEggs: 'chicken1',
      eggsPerChicken: {
        chicken1: 10,
        chicken2: 8,
      },
      eggsPerPeriod: {
        '2018-08': {
          total: 10,
          byChicken: {
            chicken1: 5,
          },
        },
      },
    };
    const wrapper = shallow(
      <FlockStatsRenderer
        stats={mockStats}
        chickens={sampleData.chickens.data}
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });
});
