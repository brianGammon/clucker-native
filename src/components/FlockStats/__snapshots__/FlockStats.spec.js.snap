// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FlockStats component: FlockStats connected component renders 1`] = `
<FlockStatsRenderer
  stats={
    Object {
      "averageNumber": 7.566666666666666,
      "averageWeight": 63.841904761904786,
      "eggsPerChicken": Object {
        "chicken1": 78,
        "chicken2": 62,
        "chicken3": 33,
        "chicken4": 49,
        "unknown": 5,
      },
      "eggsPerPeriod": Object {
        "2017-11": Object {
          "byChicken": Object {
            "chicken3": 1,
          },
          "total": 1,
        },
        "2018-02": Object {
          "byChicken": Object {
            "chicken1": 17,
            "chicken2": 14,
            "chicken3": 7,
            "unknown": 5,
          },
          "total": 43,
        },
        "2018-03": Object {
          "byChicken": Object {
            "chicken1": 25,
            "chicken2": 20,
            "chicken3": 14,
            "chicken4": 19,
          },
          "total": 78,
        },
        "2018-04": Object {
          "byChicken": Object {
            "chicken1": 24,
            "chicken2": 20,
            "chicken3": 11,
            "chicken4": 22,
          },
          "total": 77,
        },
        "2018-05": Object {
          "byChicken": Object {
            "chicken1": 12,
            "chicken2": 8,
            "chicken4": 8,
          },
          "total": 28,
        },
      },
      "firstEgg": "2017-11-16",
      "heaviest": Object {
        "chickenId": "chicken1",
        "chickenName": "Chicken1",
        "date": "2018-02-13",
        "modified": "2018-02-13T13:45:46.640Z",
        "userId": "user1",
        "weight": "73.5",
      },
      "mostEggs": "chicken1",
      "total": 227,
    }
  }
/>
`;

exports[`FlockStats component: FlockStatsRenderer renders 1`] = `
<ScrollView>
  <Component>
    {
  "total": 100,
  "heaviest": null,
  "averageWeight": 60,
  "averageNumber": 2,
  "mostEggs": "chicken1",
  "eggsPerChicken": {
    "chicken1": 10,
    "chicken2": 8
  },
  "eggsPerPeriod": {
    "2018-08": {
      "total": 10,
      "byChicken": {
        "chicken1": 5
      }
    }
  }
}
  </Component>
</ScrollView>
`;
